tensorboard/webapp/widgets/line_chart_v2/lib/series_line_view.ts

```diff
@@ -15,19 +15,140 @@ limitations under the License.
 
 import {DataDrawable} from './drawable';
 
+enum PartitionType {
+  NUMBER,
+  NAN,
+}
+
 export class SeriesLineView extends DataDrawable {
+  private partitionPolyline(
+    polyline: Float32Array
+  ): Array<{polyline: Float32Array; type: PartitionType}> {
+    if (polyline.length % 2 !== 0) {
+      throw new Error(`Cannot have odd length-ed polyline: ${polyline.length}`);
+    }
```

Comment:
Leaning not making the change. We have plenty of those invariant checks but as a dev who jumps around a lot, I think it makes sense to have the same invariant in multiple places if it is not type checkable since it makes reading the code much easier.

Reply:
In that case, can we move this check to the calling function `redraw()` ~L103?  Its function body relies on the invariant (in the loop around L137), so having the check there helps me realize that everything in `redraw()` operates with the validated assumption.